### PR TITLE
[feature] Major refactor and Iris Anim

### LINKED ISSUE #86: Change character sprite and movement animation

### PR DESCRIPTION
**Pull request by raihansyahdean**
### Main changes
- Major refactors of scripts
- Adjust scenes to new scripts (all existing scenes)
- New Character Prefab: Iris
- Animation Idle, Walk, Run for Iris

### Tasks List
- [close #86 ]

### Attachments
https://user-images.githubusercontent.com/42737788/149765764-02c8c76a-5bbb-4b03-8c49-2a87e87402ab.mp4

### New Packages
N/A

### Resources 
N/A

### CODE REVIEW

**Review by rafiftaris — CHANGES_REQUESTED** [review]
no problems in anims

File: Assets/Dev/Scripts/Audio/AudioManager.cs [hunk 1]
```diff
@@ -10,12 +8,12 @@ public class AudioManager : MonoBehaviour
...
+        Transform sound = transform.Find(name);
+        if (sound == null)
         {
             Debug.LogWarning("Sound: " + name + " not found!");
             return;
         }
-        s.GetComponent<AudioSource>().Play();
+        sound.GetComponent<AudioSource>().Play();
     }
 }
```
Comment (rafiftaris) [inline Assets/Dev/Scripts/Audio/AudioManager.cs:19]: add new line

File: Assets/Dev/Scripts/Interactables/Interactable.cs [hunk 2]
```diff
@@ -18,24 +18,21 @@ public abstract class Interactable : MonoBehaviour, IInteractable
     #region Variables
     [Header("Interactable Icons")]
     [Tooltip("True if there is an icon to be used")]
-    public bool hasIcon;
+    public bool HasIcon;
```
Comment (rafiftaris) [inline Assets/Dev/Scripts/Interactables/Interactable.cs:21]: rename this to UseIcon

File: Assets/Dev/Scripts/Player/PlayerMovement.cs [hunk 3]
```diff
@@ -17,68 +18,55 @@ public interface IPlayerMovement
...
     [Tooltip("The move direction generated")]
-    private Vector3 moveDirection;
+    private Vector3 _moveDirection;
     [Tooltip("Gravity Strength")]
-    public float gravity = -50.0f;
-    [Tooltip("Force Grounding Flag")][SerializeField]
-    private bool forceGrounding = true;
+    public float Gravity = -50.0f;
+    [Tooltip("Force Grounding Flag")] [SerializeField]
+    private bool _isForceGrounding = true;
```
Comment (rafiftaris) [inline Assets/Dev/Scripts/Player/PlayerMovement.cs:41]: rename to _useForceGrounding

**Review by rafiftaris — CHANGES_REQUESTED** [review]

File: Assets/Dev/Scripts/Audio/Sound.cs [hunk 4]
```diff
@@ -8,17 +7,17 @@
 public class Sound
```
[the remark this answers is not in the document]
Comment (raihansyahdean) [reply Assets/Dev/Scripts/Audio/Sound.cs:7]: Keep for future is needed again

**Review by rafiftaris — APPROVED** [review]
Mantappu!

**Pull request merged**